### ISSUE TITLE
SERVER_ERROR (received status code 500 and expected 200)

### ISSUE THREAD

**Issue by janhaa** [bimmerconnected/bimmer_connected#124]
```
Traceback (most recent call last):
  File "bimmer.py", line 78, in <module>
    account.update_vehicle_states()
  File "/home/install/.local/lib/python3.5/site-packages/bimmer_connected/account.py", line 215, in update_vehicle_states
    car.update_state()
  File "/home/install/.local/lib/python3.5/site-packages/bimmer_connected/vehicle.py", line 72, in update_state
    self.state.update_data()
  File "/home/install/.local/lib/python3.5/site-packages/bimmer_connected/state.py", line 130, in update_data
    params=params)
  File "/home/install/.local/lib/python3.5/site-packages/bimmer_connected/account.py", line 136, in send_request
    raise IOError(msg)
OSError: The BMW Connected Drive portal returned an error: SERVER_ERROR (received status code 500 and expected 200).
```

This occasionally happens when I try to update quite frequently (every 5 seconds or so).
Do you guys know if the API is rate-limited?

All the best

**Comment by gerard33** [comment]
There will probably be a rate limit, but I don't have any details.
Do you only get this error when lowering the frequency? Because I get this error frequently with the normal update interval of 5 minutes.

**Comment by janhaa** [comment]
I especially noticed this when increasing (I guess thats what you meant :-)) the frequency, but it might also happen at lower frequencies.
I'm wondering if it might be a problem on their end, since 500 would not be an appropiate code to return in such cases.

**Comment by gerard33** [comment]
Yeah I meant increasing the frequency (and lowering the interval) indeed :smiley:

But it's indeed a problem of the Connected Drive portal. When we started with this library this issue was not there. So it seems there is to many traffic to properly handle.

We are going to test another connection methode towards the Connected Drive Portal, see [here](https://github.com/bimmerconnected/bimmer_connected/issues/99#issuecomment-528609837), to see if that improves stuff.

**Comment by timmccor** [comment]
@janhaa Coincidentally I've submitted a PR to allow you to request the library retry any requests which result in a 500 error. See [here](https://github.com/bimmerconnected/bimmer_connected/pull/127).

This might help you whilst we await a better solution.

**Comment by timmccor** [comment]
The PR has been merged, and will be available as part of 0.6.1.

By default the library will now retry 500 errors, which are temporary server errors, 5 times with 1s delay between each retry - in all our testing this has eradicated the issue.

**Comment by janhaa** [comment]
Sweet, cheers!
